annotation/tests/test_tasks_services.py

```diff
@@ -585,3 +636,1697 @@ def test_filter_tasks_db_no_files_or_jobs(
+        1, x_current_tenant, token.token
+    )
+    mock_dependencies["mock_get_annotation_tasks"].assert_called_with(
+        mock_db, job_id, 1, {1}
+    )
+    mock_dependencies["mock_construct_annotated_pages"].assert_called_with(
+        mock_db, x_current_tenant, {}
+    )
+    mock_dependencies["mock_construct_annotated_doc"].assert_not_called()
+    mock_dependencies["mock_update_task_status"].assert_not_called()
+    mock_dependencies["mock_logger_exception"].assert_not_called()
+
+
+def test_create_validation_revisions_construct_doc_value_error(
+    mock_dependencies: Mock,
+):
+    mock_db = MagicMock()
+    x_current_tenant = "tenant_1"
+    token = MagicMock(token="fake_token")
+    job_id = 123
+    validation_tasks = [
+        ManualAnnotationTask(file_id=1, pages={1}, user_id=1, id=10)
+    ]
+
+    mock_dependencies["mock_construct_annotated_pages"].return_value = (
+        [MagicMock(objs=True)],
+        {"category1"},
+    )
+    mock_dependencies["mock_construct_annotated_doc"].side_effect = ValueError(
+        "Test error"
+    )
+
+    with patch(
+        "annotation.tasks.services.DocForSaveSchema"
+    ) as mock_doc_for_save_schema:
+        mock_doc_instance = mock_doc_for_save_schema.return_value
+        mock_doc_instance.user = validation_tasks[0].user_id
+        mock_doc_instance.pages = [MagicMock(objs=True)]
+        mock_doc_instance.validated = set()
+        mock_doc_instance.failed_validation_pages = set()
+        mock_doc_instance.categories = {"category1"}
+
+        create_validation_revisions(
+            mock_db, x_current_tenant, token, job_id, validation_tasks
+        )
+
+        mock_dependencies[
+            "mock_get_file_path_and_bucket"
+        ].assert_called_once_with(1, x_current_tenant, token.token)
+        mock_dependencies["mock_get_annotation_tasks"].assert_called_once_with(
+            mock_db, job_id, 1, {1}
+        )
+        mock_dependencies[
+            "mock_construct_annotated_pages"
+        ].assert_called_once_with(mock_db, x_current_tenant, {})
+        mock_dependencies[
+            "mock_construct_annotated_doc"
+        ].assert_called_once_with(
+            db=mock_db,
+            user_id=validation_tasks[0].user_id,
+            pipeline_id=None,
+            job_id=job_id,
+            file_id=1,
+            doc=mock_doc_instance,
+            tenant=x_current_tenant,
+            s3_file_path="s3/path",
+            s3_file_bucket="bucket",
+            latest_doc=None,
+            task_id=validation_tasks[0].id,
+            is_latest=True,
+        )
+        mock_dependencies["mock_logger_exception"].assert_called_once_with(
+            "Cannot save first validation revision."
+        )
+        mock_dependencies["mock_update_task_status"].assert_not_called()
+
+
+def test_construct_annotated_pages(mock_session: Mock):
+    x_current_tenant = "tenant_1"
+
+    with patch(
```

Comment:
There is a call to load_annotations inside of the tested function. Why wasn't it mocked in this test?